src/App.js

```diff
@@ -7,45 +7,67 @@ import { useState, useEffect } from "react";
 
 
 function App() {
-
-  
- 
   const [todoList, setTodoList] = useState([]);
-    
-    const [isLoading, setIsLoading] = useState(true);
-    
-    
-    useEffect(() => {
-      new Promise((resolve, reject) => {
-        setTimeout(() => {
-          resolve({ data: { todoList: JSON.parse(localStorage.getItem('savedTodoList')) || [], } });
-        },2000)
-      })
-      .then((result) => {
-        setTodoList(result.data.todoList);
-        setIsLoading(false);
-      });
-    },[]);
-    
-    
-    useEffect(() => {
-      if (!isLoading) {
-        localStorage.setItem('todoList', JSON.stringify(todoList));
-      }
-    }, [todoList, isLoading]);
-    
-    
-    const removeTodo = (idToRemove) => {
-      const updatedTodoList = todoList.filter((item) => item.id !== idToRemove);
-      setTodoList(updatedTodoList);
-    };
-    
-    const addTodo = (newTodo) => {
-      setTodoList([...todoList, newTodo]);
+
+  const [isLoading, setIsLoading] = useState(true);
+
+  const fetchData = async () => {
+    const options = {
+      method: "GET",
+      headers: {
+        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+      },
     };
-    
-    return (
-      <>
+
+    const url = `https://api.airtable.com/v0/${process.env.REACT_APP_AIRTABLE_BASE_ID}/${process.env.REACT_APP_TABLE_NAME}/`;
+
+    try {
+      const response = await fetch(url, options);
+
+      if (!response.ok) {
+        throw new Error(`Error: ${response.status}`);
+      }
+
+      const data = await response.json();
+
+      console.log(data);
+
+      const todos = data.records.map((todo) => ({
+        id: todo.id,
+        title: todo.fields.title,
+      }));
+      console.log(todos);
+
+      setTodoList(todos);
+      setIsLoading(false);
+    } catch (error) {
+      console.error("Error fetching data:", error.message);
+
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchData(setTodoList, setIsLoading);
```

Comment:
Since both of these functions are within the same scope (`App`) as this `useEffect` you don't need to pass them as arguments.

```suggestion
    fetchData();
```